test/unit/connectors/test_sharepoint.py

```diff
@@ -0,0 +1,135 @@
+    config.site = "https://test.sharepoint.com/sites/test"
+    config.get_client.return_value = mock_client
+    config._get_drive_item.return_value = mock_drive_item
+    return config
+
+
+@pytest.fixture
+def mock_download_config():
+    config = Mock(spec=SharepointDownloaderConfig)
+    config.max_retries = 3
+    return config
+
+
+@pytest.fixture
+def sharepoint_downloader(mock_connection_config, mock_download_config):
+    downloader = SharepointDownloader(
+        connection_config=mock_connection_config, download_config=mock_download_config
+    )
+    return downloader
+
+
+@pytest.fixture
+def file_data():
+    return FileData(
+        source_identifiers=SourceIdentifiers(
+            filename="test.docx", fullpath="/sites/test/Shared Documents/test.docx"
+        ),
+        connector_type="sharepoint",
+        identifier="test-id",
+    )
+
+
+def test_fetch_file_retries_on_429_error(
+    mock_client, mock_drive_item, mock_site, sharepoint_downloader, file_data
+):
+    """Test that _fetch_file retries when encountering 429 errors"""
+    mock_client.sites.get_by_url.return_value.get.return_value.execute_query.return_value = (
+        mock_site
+    )
+    mock_drive_item.get_by_path.return_value.get.return_value.execute_query.side_effect = [
+        Exception("429 Client Error"),
+        Exception("Request has been throttled"),
+        mock_file,
+    ]
+
+    result = sharepoint_downloader._fetch_file(file_data)
+    assert result == mock_file
+    assert mock_drive_item.get_by_path.return_value.get.return_value.execute_query.call_count == 3
+
+
+def test_fetch_file_fails_after_max_retries(
+    mock_client, mock_drive_item, mock_site, sharepoint_downloader, file_data
+):
+    """Test that _fetch_file fails after exhausting max retries"""
+    mock_client.sites.get_by_url.return_value.get.return_value.execute_query.return_value = (
+        mock_site
+    )
+    mock_drive_item.get_by_path.return_value.get.return_value.execute_query.side_effect = Exception(
+        "429 Client Error"
+    )
+
+    with pytest.raises(Exception, match="429"):
+        sharepoint_downloader._fetch_file(file_data)
+
+    expected_calls = sharepoint_downloader.download_config.max_retries
+    assert (
+        mock_drive_item.get_by_path.return_value.get.return_value.execute_query.call_count
+        == expected_calls
+    )
+
+
+def test_fetch_file_handles_site_not_found_immediately(
+    mock_client, sharepoint_downloader, file_data
+):
+    """Test that site not found errors are not retried"""
+    mock_client.sites.get_by_url.return_value.get.return_value.execute_query.side_effect = (
+        Exception("Site not found")
+    )
+
+    with pytest.raises(SourceConnectionError, match="Site not found"):
+        sharepoint_downloader._fetch_file(file_data)
+
+    assert mock_client.sites.get_by_url.return_value.get.return_value.execute_query.call_count == 1
+
+
+def test_fetch_file_without_errors(
```

Comment:
Nit: positive tests should be named just `test_<feature>`
Nit: positive tests should be at the top. There's no reason to test edge cases if the sanity checks fail.